Classes/Domain/Repository/DatabaseStorageRepository.php

```diff
@@ -67,4 +81,58 @@ public function findStorageidentifiers()
 
         return $this->identifiers;
     }
+
+    /**
+     * Delete all entries of a storage by its identifier and an optional date interval.
+     *
+     * @param string $storageIdentifier Storage identifier
+     * @param \DateInterval|null $dateInterval Date interval
+     * @return int
+     * @throws IllegalObjectTypeException
+     * @throws InvalidQueryException
+     */
+    public function deleteByStorageIdentifierAndDateInterval(string $storageIdentifier, \DateInterval $dateInterval = null): int
+    {
+        $query = $this->createQuery();
+        $constraints = [
+            $query->equals('storageidentifier', $storageIdentifier)
+        ];
+
+        if ($dateInterval !== null) {
+            $currentDate = new \DateTime();
+            $newDate = $currentDate->sub($dateInterval);
+            $constraints[] = $query->lessThan('datetime', $newDate->format('Y-m-d H:i:s'));
+        }
+
+        $query->matching($query->logicalAnd($constraints));
+        $entries = $query->execute();
+        $count = 0;
+        foreach ($entries as $entry) {
+            $this->remove($entry);
+            $count++;
+        }
+
+        return $count;
+    }
+
+    /**
+     * Get the list of all storage identifiers. Optionally exclude some.
+     * For performance reasons, this method does not use the ORM.
+     *
+     * @param array $excludedIdentifiers
+     * @return array
+     */
+    public function getStorageIdentifiers(array $excludedIdentifiers = []): array
```

Comment:
Method `findStorageIdentifiers()` should be rewritten to use this method for performance reasons, or not?